components/utilities/ai.js

```diff
@@ -0,0 +1,184 @@
+/* AI logic to play a double */
+// Need to handle logic if there are 3/4 of a kind
+// This leads to different combinations of doubles.
+const double = (hand) => {
+  const dupesObject = createDuplicatesObject(hand);
+  const combinations = [];
+
+  for (const value in dupesObject) {
+    if (dupesObject[value].length === 2) {
+      combinations.push(dupesObject[value]);
+    } else if (dupesObject[value].length > 2) {
+      for (let i = 0; i < dupesObject[value].length; i++) {
+        for (let j = i + 1; j < dupesObject[value].length; j++) {
+          combinations.push([dupesObject[value][i], dupesObject[value][j]]);
+        }
+      }
+    }
+  }
+  return combinations;
+}
+/* AI logic to play a triple */
+/** 
+ * NOTE: This solution for if the number of duplicates is four seems severely need of refactoring.
+*/
+const triple = (hand) => {
+  const dupesObject = createDuplicatesObject(hand);
+  const combinations = [];
+
+  for (const value in dupesObject) {
+    if (dupesObject[value].length === 3) {
+      combinations.push(dupesObject[value]);
+    } else if (dupesObject[value].length === 4) {
+      for (let i = 0; i < dupesObject[value].length; i++) {
+        for (let j = i + 1; j < dupesObject[value].length; j++) {
+          if (dupesObject[value][j + 1]) {
+            combinations.push([dupesObject[value][i], dupesObject[value][j], dupesObject[value][j + 1]]);
+          }
+
+          if (dupesObject[value][j + 2]) {
+            combinations.push([dupesObject[value][i], dupesObject[value][j], dupesObject[value][j + 2]]);
+          }
+        }
+      }
+    }
+  }
+  return combinations;
+}
+/* AI logic to play a quartet */
+const quartet = (hand) => {
+  const dupesObject = createDuplicatesObject(hand);
+  const combinations = [];
+
+  for (const value in dupesObject) {
+    if (dupesObject[value].length === 4) {
+      combinations.push(dupesObject[value]);
+    }
+  }
+  return combinations;
+}
+
+/* AI logic to play double sequence */
+
+
+const merge = (left, right) => {
+  let sortedArr = [];
+
+  while (left.length && right.length) {
+    if (left[0].value < right[0].value) {
+      sortedArr.push(left.shift());
+    } else {
+      sortedArr.push(right.shift());
+    }
+  }
+
+  return [...sortedArr, ...left, ...right];
+}
+
+const mergeSort = (arr) => {
+  if (arr.length <= 1) {
+    return arr;
+  }
+
+  const mid = Math.floor(arr.length / 2);
+
+  const left = mergeSort(arr.slice(0, mid));
+  const right = mergeSort(arr.slice(mid));
+
+  return merge(left, right);
+}
+
+/*
+'hearts': '♥',
+'diamonds': '♦',
+'spades': '♠',
+'clubs': '♣',
+*/
+
+const createDuplicatesObject = (array) => {
+  const values = {};
+
+  for (let i = 0; i < array.length; i++) {
+    const number = array[i].number;
+    if (!values[number]) {
+      values[number] = [array[i]];
+    } else {
+      values[number].push(array[i]);
+    }
```

Comment:
standard eslint rules prefer the if to not be a negated expression. thus you'd want to write it like

```suggestion
    if (values[number]) {
      values[number].push(array[i]);
    } else {
      values[number] = [array[i]];
    }
```